### PR TITLE
Feature/ref resolution hardening

### PR DESCRIPTION
**Pull request by aringo-bf**
## Summary

This PR improves parsing reliability for Swagger/OpenAPI specs, especially around `$ref` resolution and server/basePath handling.  

It fixes cases where request URLs were built incorrectly and where external/nested references could fail to resolve.

## Key Improvements

- Hardened external and nested `$ref` resolution across spec files
- Improved context handling and caching for external schema references
- Normalized basePath/server path handling for Swagger v2 and OpenAPI v3
- Preserved server path components when using `-T` target overrides
- Corrected handling of absolute and relative server URLs in request construction
- Added regression coverage for these edge cases

## Notes for Review

Most of the added code in this PR is test coverage and fixtures to lock in behavior and prevent regressions.  

About 74% of added lines in this commit are test-related (1159 / 1565).

Implementation changes are concentrated almost all in `cmd/utils.go` with supporting adjustments in command wiring. So review is for the most part just looking at utils.go which does the SPEC parsing. The rest of the files are the automated testing built to "make sure" that the enhanced parsing .. and really ay parsing is functioning. 

For feature/ref-resolution-hardening vs origin/main:

Total diff: +1565 / -131 (1696 changed lines)
Test-related files ([*_test.go](https://file+.vscode-resource.vscode-cdn.net/Users/aringo/.vscode/extensions/openai.chatgpt-0.4.78-darwin-arm64/webview/#) + tests/ fixtures): +1159 / -0
Go test code only ([*_test.go](https://file+.vscode-resource.vscode-cdn.net/Users/aringo/.vscode/extensions/openai.chatgpt-0.4.78-darwin-arm64/webview/#)): +723 / -0
Test fixtures only (tests/ YAML files): +436 / -0
Files changed: 16 total, 10 test-related

## Validation

- `go test ./...` passes

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -19,15 +20,21 @@ var jsonResultArray []Result
 var jsonVerboseResultArray []VerboseResult
 var specTitle string
 var specDescription string
+var externalRefCache = make(map[string]map[string]interface{})
+var specToFilePath = make(map[*interface{}]string) // Maps external spec pointers to their file paths
```
Comment (twest-bf) [inline cmd/utils.go]:
```suggestion
```
This variable is declared but never used.
